Codepipeline xc3 - Module

kashan-01 opened the pull request:
Link: https://github.com/XgridInc/xc3/issues/77

	[PS-2]
	- Feedback addressed.
	- Added some new features.
	[PS-1]
	- Crating PR of codepipeline.

Issues:
- This pipeline code is used to automate the deployment of xc3 infrastructure through codepipeline.

<br>

- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Every function, interface, class has a comment describing what it does and input/output parameters

Signed-off-by: Kashan Ali [ kashan.ali@xgrid.co ]

kashan-01 committed:
Link: https://github.com/XgridInc/xc3/issues/77
	[PS-2]
	- Added licence
<br>
	[PS-1]
	- Crating PR of codepipeline.

Issues:
- This code pipeline is used to automate the deployment of xc3 infrastructure through codepipeline.

<br>

- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Every function, interface, class has a comment describing what it does and input/output parameters

Signed-off-by: Kashan Ali [ kashan.ali@xgrid.co ]

samanXgrid commented on codepipeline-tf/modules/xc3_codepipeline/readme.md:
```diff
@@ -0,0 +1 @@
+This pipeline has currently different stages to Source, Build, Deploy and destroy the code.
```
please add all relevant detail in readme how you are deploying infra using pipeline what are the steps

samanXgrid commented on codepipeline-tf/terraform.auto.tfvars:
```diff
@@ -0,0 +1,44 @@
+##################################################
```
license is missing here

muhammad-irfan-xgrid replied:
done

samanXgrid commented on codepipeline-tf/terraform.auto.tfvars:
```diff
@@ -0,0 +1,44 @@
+s3_bucket_name = "terraform-state-xc3-example-pipeline"
+
+
+xc3_codepipeline_role  = "arn:aws:iam::201635854701:role/xccc-pipeline-role"
```
this should be dummy and value shoud be imported via sh script that runs on 1st time

kashan-01 replied:
this was for testing purpose, now we've tested roles with terraform resources and almost testing is completed. Will pass the arn of dynamically build roles on runtime.

samanXgrid commented on codepipeline-tf/terraform.auto.tfvars:
```diff
@@ -0,0 +1,44 @@
+
+xc3_codepipeline_role  = "arn:aws:iam::201635854701:role/xccc-pipeline-role"
+codebuild_service_role = "arn:aws:iam::201635854701:role/service-role/codebuild-test-service-role"
+codestar_connections   = "arn:aws:codestar-connections:eu-west-1:201635854701:connection/68b68fa1-9687-405a-b75c-cbf1f5ca6de5"
```
can we pass it via env or ssm parameter because it will be different for each repo you can mention it in pre-req thing

kashan-01 replied:
Yes, you're right. but when we create connection with resource than we'll pass the arn of role that is created on runtime.

samanXgrid reviewed (commented):
Left few comments please have a look

Commit:
Codepipeline xc3 - Module

Link: https://github.com/XgridInc/xc3/issues/77

	[PS-1]
	- Crating PR of codepipeline.

Issues:
- This code pipeline is used to automate the deployment of xc3 infrastructure through codepipeline.

<br>

- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Every function, interface, class has a comment describing what it does and input/output parameters

Signed-off-by: Kashan Ali [ kashan.ali@xgrid.co ]

Signed-off-by: kashan-ali_snlabs <kashan.ali@sleepnumber.com>

Commit:
Added new changes

Signed-off-by: kashan-ali_snlabs <kashan.ali@sleepnumber.com>

muhammad-irfan-xgrid committed:
added roles and updated the init_buildspec.yml file to bash backend.tf

muhdsaad commented on codepipeline-tf/modules/xc3_codepipeline/variables.tf:
```diff
@@ -0,0 +1,132 @@
+variable "key" {
+  description = "key used in codepipeline(state file key)."
+  type        = string
+}
```
```suggestion
}

```

muhammad-irfan-xgrid replied:
done

muhdsaad commented on codepipeline-tf/providers.tf:
```diff
@@ -0,0 +1,13 @@
+}
+
+provider "aws" {
+  region = "eu-west-1"
```
@kashan-01 We should make the region variable. And if we need eu-west-1 as default we can set it in variables.tf

muhammad-irfan-xgrid replied:
Done

muhdsaad commented on codepipeline-tf/readme.md:
```diff
@@ -0,0 +1,82 @@
+
+### Sensitive Values Variables
+
+- `xc3_codepipeline_role` (String): The IAM role used to execute the CodePipeline.
```
@kashan-01 ^^

muhdsaad reviewed (changes requested):
@kashan-01 Just a few changes are required. Thanks

muhammad-irfan-xgrid committed:
remove variables and update added command in initbuildsepc file

muhammad-irfan-xgrid committed:
updated readme and change region in provider file to make it dynamic theough avriable file

muhdsaad commented on codepipeline-tf/modules/xc3_codepipeline/readme.md:
```diff
@@ -0,0 +1 @@
+This pipeline has currently different stages to Source, Build, Deploy and destroy the code.
```
Add details on the setup and other relevant things as well.

muhammad-irfan-xgrid replied:
sure

muhdsaad reviewed (changes requested):
Please see required changes. Thanks

muhammad-irfan-xgrid committed:
updated readme for codepipelien module

SaadMohsinKhan-XCBG commented:
@muhammad-irfan-xgrid & @shehroz-khan-xgrid please resolve all conversations in this PR if those changes have been addressed.

muhammad-irfan-xgrid commented on codepipeline-tf/readme.md:
```diff
@@ -0,0 +1,82 @@
+
+- `xc3_codepipeline_role` (String): The IAM role used to execute the CodePipeline.
+
+- `codebuild_service_role` (String): The IAM role used to create projects for the CodePipeline.
```
already done

SaadMohsinKhan-XCBG reviewed (approved):
LGTM

samanXgrid commented on codepipeline-tf/main.tf:
```diff
@@ -0,0 +1,47 @@
+  s3_bucket_name              = var.s3_bucket_name
+  full_repository_id          = var.full_repository_id
+  full_branch_name            = var.full_branch_name
+  region = var.region
```
please apply terraform fmt

shehroz-khan-xgrid replied:
fixed the indentation.

samanXgrid commented on codepipeline-tf/modules/xc3_codepipeline/readme.md:
```diff
@@ -0,0 +1,89 @@
+  full_repository_id       = var.full_repository_id
+  full_branch_name         = var.full_branch_name
+  approve_comment_for_apply  = var.approve_comment_for_apply
+  approve_comment_for_destroy  = var.approve_comment_for_destroy
```
please apply tf fmt here

shehroz-khan-xgrid replied:
Done.

samanXgrid commented on codepipeline-tf/modules/xc3_codepipeline/variables.tf:
```diff
@@ -0,0 +1,132 @@
+}
+
+variable "init_buildspec" {
+  description = ""
```
We can add description here and rest of variables as well

shehroz-khan-xgrid replied:
added.

samanXgrid commented on codepipeline-tf/providers.tf:
```diff
@@ -0,0 +1,13 @@
+terraform {
+  required_version = ">= 1.0"
```
can we use latest version with proper constraints

shehroz-khan-xgrid replied:
upgraded to 1.7.

shehroz-khan-xgrid committed:
update terraform version, fix indentation and add description

shehroz-khan-xgrid committed:
remove region variable

shehroz-khan-xgrid committed:
fix region variable error by deleting it

muhammad-irfan-xgrid commented on codepipeline-tf/modules/xc3_codepipeline/providers.tf:
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.0"
```
I see in version are different in each provider file. make sure to keep same version in each provider file.

shehroz-khan-xgrid replied:
updated to 1.7, the same as the other providers have.

muhammad-irfan-xgrid commented on codepipeline-tf/providers.tf:
```diff
@@ -0,0 +1,13 @@
+terraform {
+  required_version = ">= 1.7"
```
I see in version are different in each provider file. make sure to keep same version in each provider file.

shehroz-khan-xgrid committed:
update terraform version and fix most expensive services panel issue

shehroz-khan-xgrid committed:
update terraform version and fix most expensive services panel issue

shehroz-khan-xgrid committed:
update terraform version and fix most expensive services panel issue

muhammad-irfan-xgrid reviewed (approved):
LGTM

samanXgrid reviewed (approved):
LGTM
